src/lightning/fabric/accelerators/xla.py

```diff
@@ -76,14 +76,19 @@ def auto_device_count() -> int:
                 from torch_xla._internal import tpu
 
                 return tpu.num_available_devices()
+            from torch_xla.experimental import tpu
+
             device_count_on_version = {2: 8, 3: 8, 4: 4}
             return device_count_on_version.get(tpu.version(), 8)
         return getenv_as(xenv.TPU_NUM_DEVICES, int, 8)
 
     @staticmethod
     @functools.lru_cache(maxsize=1)
     def is_available() -> bool:
-        return XLAAccelerator.auto_device_count() > 0
+        try:
+            return XLAAccelerator.auto_device_count() > 0
+        except (ValueError, AssertionError, OSError):
```

Comment:
Don't we need a comment here? These error come from torch_xla?

Reply:
Added. Yes